crates/ibc/src/applications/transfer/relay/on_recv_packet.rs

```diff
@@ -40,7 +40,10 @@ pub fn process_recv_packet_execute<Ctx: TokenTransferExecutionContext>(
             .get_channel_escrow_address(&packet.port_on_b, &packet.chan_on_b)
             .map_err(|token_err| (ModuleExtras::empty(), token_err))?;
 
-        ctx.send_coins(&escrow_address, &receiver_account, &coin)
+        ctx.send_coins_validate(&escrow_address, &receiver_account, &coin)
+            .map_err(|token_err| (ModuleExtras::empty(), token_err))?;
```

Comment:
@plafer this line is the root cause of the need for mutable context. Upon failure, we return `ModuleExtras` to perform some execution